api/org/commcare/api/persistence/SqlHelper.java

```
@@ -0,0 +1,246 @@
+            for (int i = 0; i < mPair.second.length; i++) {
+                preparedStatement.setString(i + 1, mPair.second[i]);
+            }
+            return preparedStatement;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static int insertToTable(Connection c, String storageKey, Persistable p) {
+
+        Pair<String, List<Object>> mPair = DatabaseHelper.getTableInsertData(storageKey, p);
+        PreparedStatement preparedStatement = null;
+
+        try {
+            preparedStatement = c.prepareStatement(mPair.first);
+            for (int i = 0; i < mPair.second.size(); i++) {
+                Object obj = mPair.second.get(i);
+
+                if (obj instanceof String) {
+                    preparedStatement.setString(i + 1, (String)obj);
+                } else if (obj instanceof Blob) {
+                    preparedStatement.setBlob(i + 1, (Blob)obj);
+                } else if (obj instanceof Integer) {
+                    preparedStatement.setInt(i + 1, (Integer) obj);
+                } else if (obj instanceof byte[]) {
+                    preparedStatement.setBinaryStream(i + 1, new ByteArrayInputStream((byte[])obj), ((byte[])obj).length);
+                }
+            }
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows == 0) {
+                throw new SQLException("Creating user failed, no rows affected.");
+            }
+
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    int id = generatedKeys.getInt(1);
+                    p.setID(id);
+                    return id;
+                } else {
+                    throw new SQLException("Creating user failed, no ID obtained.");
+                }
+            }
+
+        } catch (SQLException e) {
+            System.out.println("e: " + e);
```

More swallowed SQL exceptions